File: rotate-array/rotate-array.java
class Solution {
    public void rotate(int[] nums, int k) {
        int index = nums.length -k%nums.length;
        int[] last = new int[nums.length];
        int li=0;
        for(int i=index;i<nums.length ; i++){
            last[li] = nums[i];
            li++;
        }
        for(int j=0;j<index ; j++){
            last[li] = nums[j];
            li++;
        }
        for(int l=0;l<nums.length ; l++){
            nums[l] = last[l];
        }
    }
}

File: rotate-array/rotate-array.py
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        index = len(nums) - k%len(nums)
        last = nums[index:]
        first = nums[0:index]
        print(last)
        nums[:] = last +first
        return nums
            
        